lib/celluloid/proxy/abstract.rb

```diff
@@ -17,3 +17,28 @@ def __class__
     alias_method(:equal?, :==) unless instance_methods.include?(:equal?)
   end
 end
+
+class Celluloid::Proxy::AbstractCall < Celluloid::Proxy::Abstract
+  attr_reader :mailbox
+  
+  def initialize(mailbox, klass)
+    @mailbox = mailbox
+    @klass = klass
+  end
+  
+  def eql?(other)
+    @mailbox.eql?(other.mailbox) and self.__class__ == other.__class__
```

Comment:
I would suggest reversing this:

```ruby
self.__class__ == other.__class__ && @mailbox.eql?(other.mailbox)
```

Otherwise you run the risk that `other` does not respond to `#mailbox`